cas/service.c

```diff
@@ -1353,16 +1377,23 @@ static int cas_fom_tick(struct m0_fom *fom0)
 		 *
 		 */
 		if (!m0_dtm0_tx_desc_is_none(&cas_op(fom0)->cg_txd)) {
-			/*
 			struct m0_be_tx_credit dtm0logrec_cred;
+			void       *buf = NULL;
+			m0_bcount_t len = 0;
+			int	    rc;
+
+			rc = m0_xcode_obj_enc_to_buf(&M0_XCODE_OBJ(m0_cas_op_xc, cas_op(fom0)),
```

Comment:
While calculating the log record credits, we need only the length of the payload which means that we can use m0_xcode_data_size  instead of the m0_xcode_enc_to_buf call here. However, the current interface for m0_be_dtm0_log_credit() requires a pointer to the payload buffer to be passed in and the routine calculates the size itself. 
I will add a TBD here in the code and we can revisit the discussion.